### PR TITLE
AWS Lambda

### PR DESCRIPTION
**Pull request by bbelderbos**
Changes I had to make to get it working on AWS lambda

### CODE REVIEW

File: audio_program_generator/apg.py [hunk 1]
```diff
@@ -94,6 +94,7 @@ def __init__(
         to_mix: bool = False,
         sound_file: Path = None,
         attenuation: int = 0,
+        cache: bool = False,
```
Comment (jeffwright13) [inline audio_program_generator/apg.py]: No longer needed; caching removed in 1.6.0.

File: audio_program_generator/apg.py [hunk 2]
```diff
@@ -103,6 +104,7 @@ def __init__(
         self.sound_file = sound_file  # File with which to mix generated speech
         self.attenuation = attenuation  # Attenuation value, if mixing
         self.save_file = str(phrase_file.parent / phrase_file.stem) + ".mp3"
+        self.cache = cache
```
Comment (jeffwright13) [inline audio_program_generator/apg.py]: No longer needed; caching removed in 1.6.0.

File: audio_program_generator/apg.py [hunk 3]
```diff
@@ -118,14 +120,15 @@ def gen_speech(self):
...
-                speech.save(file)
+            if self.cache:
+                Path.mkdir(Path.cwd() / ".cache") if not Path(
+                    Path.cwd() / ".cache"
+                ).exists() else None
+
+                file = Path.cwd() / ".cache" / (phrase + ".mp3")
+                if not Path(file).exists():
+                    speech = gTTS(phrase)
+                    speech.save(file)
```
Comment (jeffwright13) [inline audio_program_generator/apg.py]: This code was reworked in 1.6.0 to implement in-memory "file" handling, and caching was removed. `apg.py` in 1.6.0 (branch:main) should be able to just be taken as-is. Can you please try pulling plain-vanilla 1.6.0 and checking if Lambda works with it?

Comment (bbelderbos) [reply audio_program_generator/apg.py]: ok cool, yep I will

**Comment by bbelderbos** [comment]
Closing this PR to work with the new in-memory code. I will open a new PR with code to call the AWS Lambda through AWS Gateway ...

**Pull request closed without merging**